### PR TITLE
Fix test status at data provider tests

### PR DESCRIPTION
**Pull request by martingrossmann**
# Description

A data provider test with collected assertions was in status `SKIPPED` instead of `FAILED`.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by mbeuthan — DISMISSED** [review]

File: core/src/main/java/eu/tsystems/mms/tic/testframework/execution/testng/worker/finish/MethodContextUpdateWorker.java [hunk 1]
```diff
@@ -53,8 +53,14 @@ public void onMethodEnd(MethodEndEvent event) {
         // Handle assertions and exceptions in dataprovider methods
         if (testResult.getMethod().isDataDriven()
                 && methodContext.readErrors().anyMatch(ErrorContext::isNotOptional)) {
-            testResult.setStatus(ITestResult.SKIP);
-            methodContext.setStatus(Status.SKIPPED);
+            /**
+             * Skipped is set by a failed data provider method
+             * {@see eu.tsystems.mms.tic.testframework.report.TesterraListener#onDataProviderFailure(ITestNGMethod, ITestContext, RuntimeException)}
```
Comment (Zsar) [inline core/src/main/java/eu/tsystems/mms/tic/testframework/execution/testng/worker/finish/MethodContextUpdateWorker.java:58]: To keep JavaDoc terse, consider
- `import`ing linked classes
- dropping parameters for methods without overloads

```java
/**
 * Skipped is set by a failed data provider method
 * {@see TesterraListener#onDataProviderFailure}
 */
```

File: core/src/main/java/eu/tsystems/mms/tic/testframework/execution/testng/worker/finish/MethodContextUpdateWorker.java [hunk 2]
```diff
@@ -53,8 +53,14 @@ public void onMethodEnd(MethodEndEvent event) {
         // Handle assertions and exceptions in dataprovider methods
         if (testResult.getMethod().isDataDriven()
                 && methodContext.readErrors().anyMatch(ErrorContext::isNotOptional)) {
-            testResult.setStatus(ITestResult.SKIP);
-            methodContext.setStatus(Status.SKIPPED);
+            /**
```
Comment (Zsar) [inline core/src/main/java/eu/tsystems/mms/tic/testframework/execution/testng/worker/finish/MethodContextUpdateWorker.java:56]: JavaDoc has to be outside method. Suggestion: Move whole comment to declaration.

File: core/src/main/java/eu/tsystems/mms/tic/testframework/report/TesterraListener.java [hunk 3]
```diff
@@ -73,7 +73,6 @@
 /**
  * Testerra listener based on TestNG for whole test execution.
  *
```
Comment (Zsar) [inline core/src/main/java/eu/tsystems/mms/tic/testframework/report/TesterraListener.java:75]: Stray linebreak left behind after removing `@author`.

File: core/src/main/java/eu/tsystems/mms/tic/testframework/report/TesterraListener.java [hunk 4]
```diff
@@ -484,6 +483,9 @@ public void onDataProviderFailure(ITestNGMethod testNGMethod, ITestContext testC
             } else {
                 methodContext.addError(exception);
             }
+            // A crashed dataprovider method as a kind of setup occurs a skipped status at the test method
```
Comment (Zsar) [inline core/src/main/java/eu/tsystems/mms/tic/testframework/report/TesterraListener.java:486]: "[D]ata[P]rovider" as per `@DataProvider` or "data[P]rovider" as per `@Test#dataProvider`.
... I fear I do not understand the rest of the sentence, but I am sure _something_ is wrong. Please have a second look.

**Review by Zsar — APPROVED** [review]

File: core/src/main/java/eu/tsystems/mms/tic/testframework/execution/testng/worker/finish/MethodContextUpdateWorker.java (hunk 2 above)
Comment (martingrossmann) [reply core/src/main/java/eu/tsystems/mms/tic/testframework/execution/testng/worker/finish/MethodContextUpdateWorker.java:56]: Simplified the comment

File: core/src/main/java/eu/tsystems/mms/tic/testframework/report/TesterraListener.java (hunk 4 above)
Comment (martingrossmann) [reply core/src/main/java/eu/tsystems/mms/tic/testframework/report/TesterraListener.java:486]: Updated comment.

**Pull request merged**